### PR TITLE
[Docs] Translate yolov5_tutorial.md

### PR DESCRIPTION
**Pull request by xin-li-67**
## Motivation

Translate yolov5 tutorial.

## Modification

 - docs/en/user_guides/yolov5_tutorial.md

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/user_guides/yolov5_tutorial.md [hunk 1]
```diff
@@ -1 +1,103 @@
...
+
+For more detailed information about environment configuration, please refer to [get_started](../get_started.md).
+
+## Dataset Preparation
+
+In this tutorial, the training dataset for MMYOLO is less than 40MB and is selected from the balloon dataset.
+
+```shell
+python tools/misc/download_dataset.py  --dataset-name balloon --save-dir data --unzip
+python tools/dataset_converters/balloon2coco.py
```
Comment (PeterH0323) [inline docs/en/user_guides/yolov5_tutorial.md:35]:
```suggestion
python tools/misc/download_dataset.py  --dataset-name balloon \
                                                                  --save-dir data \
                                                                  --unzip
python tools/dataset_converters/balloon2coco.py
```

File: docs/en/user_guides/yolov5_tutorial.md [hunk 2]
```diff
@@ -1 +1,103 @@
...
+```
+
+After executing the above command, the balloon dataset will be downloaded in the `data` folder with the converted format we need. The `train.json` and `val.json` are the annotation files; both are in the coco format.
```
Comment (PeterH0323) [inline docs/en/user_guides/yolov5_tutorial.md:38]:
```suggestion
After executing the above command, the balloon dataset will be downloaded in the `data` folder with the converted format we need. The `train.json` and `val.json` are the annotation files, both are in the coco format.
```

**Review by PeterH0323 — CHANGES_REQUESTED** [review]

File: docs/en/user_guides/yolov5_tutorial.md [hunk 3]
```diff
@@ -1 +1,249 @@
...
+cd mmyolo
+python tools/train.py configs/yolov5/yolov5_s-v61_syncbn_fast_1xb4-300e_balloon.py \
+                      --cfg-options load_from='yolov5_s-v61_syncbn_fast_8xb16-300e_coco_20220918_084700-86e02187.pth' model.backbone.frozen_stages=4 custom_hooks=None
+```
+
+### Visualization
+
+For `visualization` of `default_hooks` in `configs/yolov5/yolov5_s-v61_syncbn_fast_1xb4-300e_balloon.py`, we set `draw` to `True` and `interval` to `2`.
+
+```shell
```
Comment (PeterH0323) [inline docs/en/user_guides/yolov5_tutorial.md:151]:
```suggestion
```python
```

**Review by PeterH0323 — CHANGES_REQUESTED** [review]

File: docs/en/user_guides/yolov5_tutorial.md [hunk 4]
```diff
@@ -1 +1,249 @@
-# From Getting Started to Deployment tutorial with YOLOv5
+# YOLOv5: From Start to Deploy
```
Comment (hhaAndroid) [inline docs/en/user_guides/yolov5_tutorial.md:1]: From getting started to deployment with YOLOv5

File: docs/en/user_guides/yolov5_tutorial.md [hunk 5]
```diff
@@ -1 +1,249 @@
...
+```
+
+For more detailed information about environment configuration, please refer to [get_started](../get_started.md).
+
+## Dataset Preparation
+
+In this tutorial, the training dataset for MMYOLO is less than 40MB and is selected from the balloon dataset.
+
+```shell
+python tools/misc/download_dataset.py  --dataset-name balloon \
```
Comment (hhaAndroid) [inline docs/en/user_guides/yolov5_tutorial.md:34]: python tools/misc/download_dataset.py  --dataset-name balloon --save-dir data --unzip

File: docs/en/user_guides/yolov5_tutorial.md [hunk 6]
```diff
@@ -1 +1,249 @@
...
+
+If you don't set `custom_hooks=None` during training, the test command should be as follows:
+
+```shell
+python tools/test.py configs/yolov5/yolov5_s-v61_syncbn_fast_1xb4-300e_balloon.py \
+                      work_dirs/yolov5_s-v61_syncbn_fast_1xb4-300e_balloon/epoch_300.pth \
+                      --show-dir show_results
+```
+
+Run the above command, the inference result picture will be automatically saved to the `work_dirs/yolov5_s-v61_syncbn_fast_1xb4-300e_balloon/{timestamp}/show_results` folder. The following is one of the result pictures. Left picture is the actual annotation, and the right picture is the model inference result.
```
Comment (cydiachen) [inline docs/en/user_guides/yolov5_tutorial.md:241]: The left and the right

File: docs/en/user_guides/yolov5_tutorial.md [hunk 7]
```diff
@@ -1 +1,249 @@
...
+mim install -v -e .
+# "-v" means verbose, or more output
+# "-e" means install the project in editable mode, so any local modifications made to the code will take effect, eliminating the need to reinstall.
+```
+
+For more detailed information about environment configuration, please refer to [get_started](../get_started.md).
+
+## Dataset Preparation
+
+In this tutorial, the training dataset for MMYOLO is less than 40MB and is selected from the balloon dataset.
```
Comment (cydiachen) [inline docs/en/user_guides/yolov5_tutorial.md:31]: In this tutorial, we provide the ballon dataset, which is less than 40MB, as the training dataset for MMYOLO.

File: docs/en/user_guides/yolov5_tutorial.md [hunk 8]
```diff
@@ -1 +1,249 @@
...
+
+```shell
+python tools/misc/download_dataset.py  --dataset-name balloon \
+                                                                  --save-dir data \
+                                                                  --unzip
+python tools/dataset_converters/balloon2coco.py
+```
+
+After executing the above command, the balloon dataset will be downloaded in the `data` folder with the converted format we need. The `train.json` and `val.json` are the annotation files, both are in the coco format.
```
Comment (cydiachen) [inline docs/en/user_guides/yolov5_tutorial.md:40]: The `train.json` and `val.json` are the annotation files in the coco format.

File: docs/en/user_guides/yolov5_tutorial.md [hunk 9]
```diff
@@ -1 +1,249 @@
...
+
+test_evaluator = val_evaluator
+
+model = dict(bbox_head=dict(head_module=dict(num_classes=1)))
+
+default_hooks = dict(logger=dict(interval=1))
+```
+
+The above configuration is inherited from `./yolov5_s-v61_syncbn_fast_8xb16-300e_coco.py`, and `data_root`, `metainfo`, `train_dataloader`, `val_dataloader`, `num_classes` and other configurations are updated according to the balloon data we are using.
+We set the `interval` of the logger to 1 because each iteration of the `interval` will output a loss-related log, and the balloon dataset we use is relatively tiny. We will not see the loss-related output if the `interval` is too large.
```
Comment (cydiachen) [inline docs/en/user_guides/yolov5_tutorial.md:93]:
```suggestion
The reason why we set the `interval` of the logger to 1 is that the balloon data set we choose is relatively small, and if the `interval` is too large, we will not see the output of the loss-related log. Therefore, by setting the `interval` of the logger to 1 will ensure that each interval iteration will output a loss-related log.
```
You can decide by your own.

File: docs/en/user_guides/yolov5_tutorial.md [hunk 10]
```diff
@@ -1 +1,249 @@
...
+
+If training stops midway, add `--resume` at the end of the training command, and the program will automatically load the latest weight file from `work_dirs` to resume training.
+
+```shell
+python tools/train.py configs/yolov5/yolov5_s-v61_syncbn_fast_1xb4-300e_balloon.py --resume
+```
+
+### Fine-tune with loaded pretrained weights
+
+Our testing shows that with the YOLOv5-s pretrained model to train and validate, the coco/bbox_mAP can increase by more than 30 percent on the balloon dataset compared with not using the pre-training model solution.
```
Comment (cydiachen) [inline docs/en/user_guides/yolov5_tutorial.md:117]: NOTICE: It is highly recommended that finetuning from large datasets, such as COCO, can significantly boost the performance of overall network. 
In this example, compared with training from scratch, finetuning the pretrained model outperforms with a significant margin. (Over 30+ mAP boost than training from scratch).

File: docs/en/user_guides/yolov5_tutorial.md [hunk 11]
```diff
@@ -1 +1,249 @@
...
+For `visualization` of `default_hooks` in `configs/yolov5/yolov5_s-v61_syncbn_fast_1xb4-300e_balloon.py`, we set `draw` to `True` and `interval` to `2`.
+
+```python
+default_hooks = dict(
+     logger=dict(interval=1),
+     visualization=dict(draw=True, interval=2),
+)
+```
+
+Then re-run the following training command. During the validation, each `interval` image will save a puzzle of the annotation and prediction results to `work_dirs/yolov5_s-v61_syncbn_fast_1xb4-300e_balloon/{timestamp}/vis_data/vis_image` folder.
```
Comment (cydiachen) [inline docs/en/user_guides/yolov5_tutorial.md:158]: Re-run

**Comment by xin-li-67** [comment]
@hhaAndroid @cydiachen Thanks for the corrections. Please take a look at my latest version~

**Review by hhaAndroid — APPROVED** [review]
